### PR TITLE
Submit Elemelons

### PR DESCRIPTION
**Pull request by AlphaKretin**
I'll comment with the card texts later

### CODE REVIEW

**Comment by AlphaKretin** [comment]
Crimson Elemelon Flamelon
FIRE Plant/Effect Level 3
1800/200
The effects of your opponent's monsters in the same column as this card are negated, except for the effects of FIRE monsters. This card can make a number of attacks in each Battle Phase, up to the number of monsters with different Attributes you control in your Main Monster Zones. Once per turn: you can destroy 1 "Elemelon" monster you control, and if you do, add 1 "Elemelon" monster with a different Attribute from that destroyed card from your Deck to your hand. If this card is destroyed by battle or by card effect and sent to the GY: you can banish 1 "Elemelon" monster from your GY with a different Attribute; Special Summon this card from the GY.

Icebox Elemelon Watermelon
WATER
200/1800
The effects of your opponent's monsters in the same column as this card are negated, except for WATER monsters. When this card is Summoned: you can destroy 1 "Elemelon" monster of a different Attribute in your Deck. Once per turn: you can destroy 1 "Elemelon" monster in your hand, and if you do, add 1 "Elemelon" Spell/Trap card from your Deck to your hand. If this card is destroyed by battle or by card effect and sent to the GY: you can banish 1 "Elemelon" monster from your GY with a different Attribute; Special Summon this card from the GY.

Superstar Elemelon Rockmelon
EARTH
1500/500
The effects of your opponent's monsters in the same column as this card are negated, except for EARTH monsters. When you Normal Summon an "Elemelon" monster: You can Special Summon this card from your hand. Once per turn, if an "Elemelon" monster is banished from your GY while you control this card: you can add an "Elemelon" Spell/Trap Card from your GY to your hand. If this card is destroyed by battle or by card effect and sent to the GY: you can banish 1 "Elemelon" monster from your GY with a different Attribute; Special Summon this card from the GY.

Pixie Elemelon Electlon
WIND
600/1400
The effects of your opponent's monsters in the same column as this card are negated, except for WIND monsters. You can Normal Summon 1 "Elemelon" monster in addition to your Normal Summon/Set this turn. (You can only gain this effect once per turn.) Once per turn: you can target 1 monster in the same column as an "Elemelon" monster you control with a different Attribute from this card; destroy this card, and if you do, return that target to the hand. If this card is destroyed by battle or by card effect and sent to the GY: you can banish 1 "Elemelon" monster from your GY with a different Attribute; Special Summon this card from the GY.

Jubilee Elemelon Prismelon
LIGHT
1100/900
The effects of your opponent's monsters in the same column as this card are negated, except for LIGHT monsters. Once per turn: you can destroy this card, and if you do, shuffle a number of "Elemelon" cards from your GY into your Deck, up to the number of "Elemelon" monsters with different Attributes you control, then all "Elemelon" monsters you control gain ATK and DEF equal to the number of "Elemelon" monsters with different attributes you returned to the Deck x 300 until the end of the turn. If this card is destroyed by battle or by card effect: you can banish 1 "Elemelon" monster from your GY with a different Attribute; Special Summon this card from the GY. If an "Elemelon" monster with a different Attribute from this card is Special Summoned from your GY to a Main Monster Zone while this card is in the GY: you can Special Summon this card, but banish it when it leaves the field. You can only use this effect of "Jubilee Elemelon Prismelon" once per turn.

Black Elemelon Diamelon
DARK Plant/SpSummon/Effect Level 6
0/2500
Cannot be Normal Summoned/Set. You can Special Summon this card (from your hand) by banishing 2 "Elemelon" monsters with different Attributes from your GY. Negate the effects of your opponent's monsters in the same column as this card, except DARK monsters. Thrice per turn, if an "Elemelon" monster is destroyed while you control this card: you can add 1 "Elemelon" card from your Deck to your hand. If this card is destroyed by battle or by card effect and sent to the GY: you can banish 1 "Elemelon" monster from your GY with a different Attribute; Special Summon this card from the GY.

Elemelon-tal Farmistress Doriado
EARTH Spellcaster/Effect Level 4
1200/1400
This card is also FIRE, WIND, WATER, and LIGHT Attributes on the field. When this card is Summoned: You can activate 1 of these effects.
● You can destroy a number of Spell/Trap Cards on the field, up to the number of "Elemelon" monsters you control with different Attributes, except this card.
● Add 1 FIRE, WATER, WIND, or EARTH Plant-type monster from your Deck to your hand.

Doriado's Elemelon Farm
Field Spell
All "Elemelon" monsters you control gain 1000 ATK and DEF. When this card is activated: you can add 1 "Elemelon" card from your Deck to your hand. This card gains the following effects depending on how many "Elemelon" monsters you control.
●1+: Each time an "Elemelon" monster(s) is destroyed by battle or card effect, immediately gain 1000 LP.
●2+: Monster effects in your opponent's hand cannot be activated.
●3+: Monster effects in your opponent's GY, and the effects of your opponent's banished monsters, cannot be activated.
●4+: Your opponent cannot set Spell or Trap Cards.
●5+: Your opponent cannot activate Spell or Trap Cards.

Seedless Elemelon
Quick-Play Spell
Pay 1000 LP; Special Summon 1 "Elemelon" monster from your Deck or GY, but its ATK and DEF become 0 and it is banished during the End Phase.

Elemelon Tiller
Continuous Spell
Once per turn, this card cannot be destroyed by card effects. Each time an "Elemelon" monster is Summoned or an "Elemelon" monster's effect activates, place 1 Tiller Counter on this card. During either player's turn: you can remove 1 Tiller Counter from this card; move 1 "Elemelon" monster you control to an unoccupied Main Monster Zone you control. You can remove 3 Tiller Counters from this card, then target 1 banished "Elemelon" monster; return it to its owner's GY.

Elemelon Harvest Season!
Normal Trap
During your opponent's turn, when an "Elemelon" monster you control is destroyed by battle or card effect: you can Special Summon a number of "Elemelon" monsters with different attributes from the GY, up to the number of monsters your opponent controls.

Elemelon Melobliteration
Counter Trap
When your opponent activates a card or effect in the same column as an "Elemelon" monster you control: negate that effect, then you can destroy cards on the field up to the number of "Elemelon" monsters with different Attributes you control. You can only activate 1 "Elemelon Melobliteration" per turn.

**Comment by Gideon9212** [comment]
:drooling_face: Melons...
The column effects are nice. Each requires you to think on placement and your opponents type. Their stats aren't too high without support which may have to be nerf. Lets take a look.

> If this card is destroyed by battle or by card effect and sent to the GY: you can banish 1 "Elemelon" monster from your GY with a different Attribute; Special Summon this card from the GY.

Should OPT lock this. Not hard lock though. I can see float issues if you don't. (Using necroface or other cards to refresh)

Now going down from ID
-Crimson Elemelon Flamelon
"This card can make a number of attacks"
I like this but consider discounting itself. 
"Once per turn: you can destroy 1 "Elemelon"" I like this but consider making it so the elemelon cannot activate its "If this card is destroyed by battle or by card effect " effect.

-Icebox Elemelon Watermelon
"Once per turn: you can destroy 1 "Elemelon" monster in your hand, " Love it. Same as above though.... maybe. 

-Superstar Elemelon Rockmelon
Should consider making its def higher since its earth. Atk can stay. (Maybe up to 1000)
"Once per turn, if an "Elemelon" monster is banished from your GY while you control this card: you can add an "Elemelon" Spell/Trap Card from your GY to your hand. "
This quite a strong recovery considering. However I think its fine.

-Pixie Elemelon Electlon
Extra normal summon is fine.
The destroy this card and return is fine. 

-Jubilee Elemelon Prismelon
"Once per turn: you can destroy this card, and if you do, shuffle a number of "Elemelon" cards"
Since you have spell/trap recovery on superstar, you should make this for monsters only. If so, boost the atk and def gain to 400.
"If an "Elemelon" monster with a different Attribute from this card is Special Summoned from your GY to a Main Monster Zone while this card is in the GY: you can Special Summon this card, but banish it when it leaves the field. You can only use this effect of "Jubilee Elemelon Prismelon" once per turn."
Hmm. I'm assuming you added this as a easy way to use its effect. I see a easy way to abuse this in a xyz lv3 plant deck with only this and pixie, then staple plants. Consider making so you cannot extra deck summon that turn as well.

-Black Elemelon Diamelon
Woo the boss as a support with search. Interesting.
Good chance this won't be on the field long to use it 3 times. We will see in time. This should be fine.

-Elemelon-tal Farmistress Doriado
Non-plant so it can't be searched with the others and low stats but good recovery. Good balance here.

-Doriado's Elemelon Farm
Ok heres where a nerf maybe need or move around.
"●2+: Monster effects in your opponent's hand cannot be activated."
Consider making this 
"Once per turn: If your opponent activates a monster effect from their hand; negate it and send it to the graveyard"
Allows counterplay if they dependent on hand traps if they have more then one.
"●3+: Monster effects in your opponent's GY, and the effects of your opponent's banished monsters, cannot be activated."
Consider making this
"Negate Monster effects in your opponent's GY and banished zone."
This allows some counter play while still providing a flood gate.

"●5+: Your opponent cannot activate Spell or Trap Cards."
So all monster zone monster effects are negated and you cannot activate spell or trap cards, or monsters in hand or grave or banish....I see a problem with that. Consider making this "Your opponent cannot activate spells"
This allows if they set a trap before you placed 4, to react, providing a way out though small.

-Seedless Elemelon
Due to the s/t recovery and the field, consider making the payment 1500 LP.

-Elemelon Tiller
I like the chance of movement. Consider making the max counters it can hold is 6. This way people can't just keep holding counters and feel they must make a move.

-Elemelon Harvest Season!
Love it, however consider hard OPT locking this.

-Elemelon Melobliteration
Oh hey hard OPT locking. XD 
Love it. Should be fine.

I'll be waiting for a replay or a cdb change.

**Comment by AlphaKretin** [comment]
Replying in order, any changes I don't comment on I agree to:
You can't soft-OPT a float, they reset when the card leaves the field, but I'd agree to HOPT locking them.

I can't remember if I already discount Flamelon's self in it's multi-attack effect script-wise, but either way can update it and the text to reflect that. I also need to reword it to reflect the fact that a monster with multiple attributes counts multiple times, unless you want me to change that. Activating the floats of the other Melons is part of the point of the highlighted effects of Flamelon and Watermelon... is that maybe more acceptable since those effects are becoming Hard OPT? 

Jubilee's effect in grave is meant for Xyz, so completely locking out Extra summoning would be a slight issue. It was inspired by Chiwen, Light of the Yang Zing if you want comparison/precedent from official cards for effects like these. I'd be open to a restriction on summoning that turn though, maybe... only Plant types? Is there more than one good Plant type Xyz? I dunno... But something.

You didn't have a problem with Diamelon, but I realised after the fact and am writing as a reminder to myself that it should be *Up to* 3 times per turn, is I think proper PSCT. (I also need to use (Quick Effect) since I'm using GY...)

Amassing a huge amount of counters on Tiller is fun though...! :V Jokes and bias aside, I see your point.

Aaaaand that's it! I'll make the undisputed changes next time I get a change (which'll be tomorrow), and we can discuss Flamelon and Prismelon over DMs or something.

**Comment by Gideon9212** [comment]
Right, wasn't thinking there on the soft OPT. :rofl: 

> I can't remember if I already discount Flamelon's self in it's multi-attack effect script-wise, but either way can update it and the text to reflect that. I also need to reword it to reflect the fact that a monster with multiple attributes counts multiple times, unless you want me to change that. Activating the floats of the other Melons is part of the point of the highlighted effects of Flamelon and Watermelon... is that maybe more acceptable since those effects are becoming Hard OPT?

I think considering its going to a hard OPT, you can leave the float effects as is. Ignore me on that one sorry.

> Jubilee's effect in grave is meant for Xyz, so completely locking out Extra summoning would be a slight issue. It was inspired by Chiwen, Light of the Yang Zing if you want comparison/precedent from official cards for effects like these. I'd be open to a restriction on summoning that turn though, maybe... only Plant types? Is there more than one good Plant type Xyz? I dunno... But something.

Point made. Ok. How about
"During this turn, you must pay 1000 when a monster is summoned from the extra deck." Fits the theme and prevents abuse.

> You didn't have a problem with Diamelon, but I realised after the fact and am writing as a reminder to myself that it should be Up to 3 times per turn, is I think proper PSCT. (I also need to use (Quick Effect) since I'm using GY...)

The current wording is fine but saying 3 is less confusing to people.

**Comment by AlphaKretin** [comment]
Sounds good to me. Just trying to think how to word that.

```If an "Elemelon" monster with a different Attribute from this card is Special Summoned from your GY to a Main Monster Zone while this card is in the GY: you can Special Summon this card, but banish it when it leaves the field, also for the rest of this turn you must pay 1000 LP to Special Summon a monster from the Extra Deck.```
or
```If an "Elemelon" monster with a different Attribute from this card is Special Summoned from your GY to a Main Monster Zone while this card is in the GY: you can Special Summon this card, but apply the following effects.
● Banish this card when it leaves the field.
● For the rest of this turn, you must pay 1000 LP to Special Summon a monster from the Extra Deck.```

**Comment by AlphaKretin** [comment]
Noting feedback for myself when I go to do this tomorrow: First option, also make it "monster**(s)**", maybe edit a shovel into doriado's pic.

**Comment by AlphaKretin** [comment]
I *think* that's all the changes discussed, obvs say so if I've missed something.

**Comment by Gideon9212** [comment]
Scripts are ok. CDB is fine. This will be now merged.
